src/polyfem/assembler/GenericProblem.cpp

```diff
@@ -80,6 +86,130 @@ namespace polyfem
 				form_ = 0;
 		}
 
+		double PiecewiseConstantInterpolation::eval(const double t) const
+		{
+			if (t < points_[0])
+				return 0;
+			for (size_t i = 0; i < points_.size()-1; ++i) {
+				if (t >= points_[i] && t < points_[i+1]) {
+					return values_[i];
+				}
+			}
+			if (extend_ == 0)
```

Comment:
This is the same between all `PiecewiseInterpolation` classes, right? If so move it into a function in the `PiecewiseInterpolation` parent class and call that one in the child class when needed.